django/contrib/postgres/fields/array.py

```diff
@@ -88,6 +88,16 @@ def to_python(self, value):
             value = [self.base_field.to_python(val) for val in vals]
         return value
 
+    def from_db_value(self, value, expression, connection, context):
+        if value is None:
+            return value
+        return [
+            self.base_field.from_db_value(val, expression, connection, context)
+            if hasattr(self.base_field, 'from_db_value') else
```

Comment:
Correct - there is a significant performance improvement when no fields need a converter (defined in the db backend or as `from_db_value`). Arguably it would be nicest if `ArrayField` only defined it when it was needed, but this would cause inheritance issues so I'm happy for it to be there all the time.